BUILD.gn

```diff
@@ -59,6 +59,7 @@ if (current_toolchain != "${dir_pw_toolchain}/dummy:dummy") {
       deps += [
         "${chip_root}/src/qrcodetool",
         "${chip_root}/src/setup_payload",
+        "${chip_root}/examples/shell",
```

Comment:
is "shell" an example, or more like one of the tools?  it could reasonably live in src? I'm interested in @gerickson's feedback here.

Reply:
The intention is for `chip-shell` app to be portable code that can run on-device on embedded systems, and also serve as a reasonable tool in the standalone / linux context for simulation.

Reply:
in that case, I think CHIP would continue to maintain it in-repo and it can live with chiptool, et al.

Reply:
`chip-tool` is in `examples` too. :)  
Either way, let's decouple any restructuring proposals from this PR.

Reply:
> is "shell" an example, or more like one of the tools? it could reasonably live in src? I'm interested in @gerickson's feedback here.

This (the shell) seems to be an interesting animal. It's sort of a diagnostic example app rather than an administrative tool. I can see an argument for living under `examples/` and an argument for living under `src/`.